File: app/models/spotify_cred.rb
# == Schema Information
#
# Table name: spotify_creds
#
#  id            :integer          not null, primary key
#  access_token  :string
#  refresh_token :string
#  expires_at    :string
#  created_at    :datetime         not null
#  updated_at    :datetime         not null
#  user_id       :integer
#  spotify_id    :string
#  playlist_id   :string
#

class SpotifyCred < ActiveRecord::Base
  belongs_to :user
  validates :access_token, presence: true
  validates :refresh_token, presence: true
  validates :expires_at, presence: true
<<<<<<< HEAD
<<<<<<< Updated upstream
=======
=======
>>>>>>> 52eb835b94149da674d82ff7891b3993f3456708
  validates :spotify_id, presence: true

  def update_playlist(playlist_id)
      self.playlist_id = playlist_id
      self.save
  end
<<<<<<< HEAD

  def update_username(username)
      if (username)
          self.spotify_id = username
          return true
      end
      return false
  end
>>>>>>> Stashed changes
=======
>>>>>>> 52eb835b94149da674d82ff7891b3993f3456708
end

File: src/app/tempo/models/spotify_cred.py
from . import *

class SpotifyCred(Base):
  __tablename__ = 'spotify_creds'

  id = db.Column(db.Integer, primary_key = True)
  user_id = db.Column(db.Integer, db.ForeignKey('users.id', ondelete = 'CASCADE'))
  access_token = db.Column(db.String, nullable = False)
  refresh_token = db.Column(db.String, nullable = False)
  expires_at = db.Column(db.String, nullable = False)
  spotify_id = db.Column(db.String)
  playlist_id = db.Column(db.String)

  def __init__(self, **kwargs):
    self.user_id = kwargs.get('user_id')
    self.access_token = kwargs.get('access_token')
    self.refresh_token = kwargs.get('refresh_token')
    self.expires_at = kwargs.get('expires_at')
    self.spotify_id = kwargs.get('spotify_id', None)
    self.playlist_id = kwargs.get('playlist_id', None)
